charts/kong/templates/_helpers.tpl

```diff
@@ -365,10 +365,12 @@ The name of the service used for the ingress controller's validation webhook
   emptyDir: {}
 - name: {{ template "kong.fullname" . }}-tmp
   emptyDir: {}
+{{- if (and (eq .Values.env.database "postgres") .Values.waitImage.enabled) }}
```

Comment:
The logic, as I understand it, is about "database being present", not about "database being of type postgres". Therefore, `!= off` rather than `== "postgres"` feels more appropriate to me. Please change the logic to the former unless there's a compelling reason not to do so.

(applies to other similar checks "if database is postgres" in this PR as well)

Reply:
There's a reason it's like this, though the check is a bit off. These waits, for the initial migrations, are only useful if Postgres being online or not is influenced by the chart lifecycle. This is only true if you use the sub-chart and Postgres spawns alongside the migration Jobs.

If you manage Postgres or Cassandra independently (all Cassandra instances are managed independently--we have no sub-chart for it), the chart lifecycle has no influence on whether the database is available or not. It either is, and migrations can start immediately, or it isn't, and they cannot. No amount of waiting helps if an independently-managed database isn't online; it makes sense to just let the failure happen immediately so users know they need to take action to make the database reachable. e5012beea782a95686f9256f65ba373dda966a74 is more correct for these, since it changes the check from "any Postgres" to "only sub-chart Postgres".

The Deployment waits, however, are to see if migrations have completed. Those apply to any database and are always determined by the chart lifecycle. I did, however, copy the wrong logic when adding the `waitImage.enabled` for them :facepalm: 68bd5ffe3fb2b4ae0df82a9d540fc7cbd782fb1c fixes that.